packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx

```diff
@@ -378,28 +376,23 @@ export class InputTimeZone
    * @param open
    * @private
    */
-  private overrideSelectedLabelForRegion(open: boolean): void {
+  private overrideSelectedLabelForRegion(): void {
     if (this.mode !== "region" || !this.selectedTimeZoneItem) {
       return;
     }
 
-    const { label, metadata } = this.selectedTimeZoneItem;
-
-    this.comboboxEl.selectedItems[0].textLabel =
-      !metadata.country || open
-        ? label
-        : getSelectedRegionTimeZoneLabel(label, metadata.country, this.messages);
+    this.comboboxEl.selectedItems[0].textLabel = this.getItemLabel(this.selectedTimeZoneItem);
   }
 
   private onComboboxBeforeClose(event: CustomEvent): void {
     event.stopPropagation();
-    this.overrideSelectedLabelForRegion(false);
```

Comment:
Can you double check this doesn't cause any label update flicker on the selected item? For context, this was needed before to make set the future label before rendering, which depends on the open state.

Reply:
@josercarcamo I just tested locally and noticed flickering when the item’s label updates. You can see this regression by comparing [before](https://codepen.io/jcfranco/pen/GgJdRXM) and [after](https://codepen.io/jcfranco/pen/PwqeodE?editors=1010).

Looks like we’ll need to revert overrideSelectedLabelForRegion to the previous version:

1.	https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx#L381-L392
2.	https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/src/components/input-time-zone/input-time-zone.tsx#L394-L404

Can you take a look?